tiny_skia/src/backend.rs

```diff
@@ -563,11 +558,11 @@ impl Backend {
                     clip_mask,
                     clip_bounds,
                     scale_factor,
-                    translation + *offset,
+                    transformation * *new_transformation,
                 );
             }
             Primitive::Clip { bounds, content } => {
-                let bounds = (*bounds + translation) * scale_factor;
+                let bounds = (*bounds * transformation) * scale_factor;
```

Comment:
This line is incorrect. Since transformation already can contain scaling, `scale_factor` here is redundant and causes clipping issues (see the P.S in https://discourse.iced.rs/t/rendering-way-too-many-rectangles-on-a-canvas/113/12?u=0x7cfe).